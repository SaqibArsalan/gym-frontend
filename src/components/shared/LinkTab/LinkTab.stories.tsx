// Generated with util/create-component.js
import React from 'react';
import LinkTab from './LinkTab';

export default {
	title: 'LinkTab',
};

export const WithIcon = () => (
	<LinkTab label='With icon'  href='#' />
);

export const WithoutIcon = () => <LinkTab label='No icon' href='#' />;
