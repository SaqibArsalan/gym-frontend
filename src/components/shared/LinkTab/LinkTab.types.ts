/* eslint-disable no-unused-vars */
import React from 'react';

// Generated with util/create-component.js
// eslint-disable-next-line no-shadow
export enum IconPositionEnum {
	top = 'top',
	start = 'start',
	end = 'end',
	bottom = 'bottom',
}

export interface ILinkTabProps {
	label: string;
	href: string;
	onClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
	children?: any;
	icon?:
		| string
		| React.ReactElement<any, string | React.JSXElementConstructor<any>>;
	iconPosition?: IconPositionEnum;
	value?: number;
}
