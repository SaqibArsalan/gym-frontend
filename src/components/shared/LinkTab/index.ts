export { default as LinkTab } from './LinkTab';
export { IconPositionEnum } from './LinkTab.types';
