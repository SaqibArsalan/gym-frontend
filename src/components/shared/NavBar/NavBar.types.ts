/* eslint-disable no-unused-vars */
import React from 'react';
import { TabsProps } from '@mui/material';
// Generated with util/create-component.js
export interface INavBarProps extends TabsProps {
	children: React.ReactNode;
	removeBorderBottom?: boolean;
}
