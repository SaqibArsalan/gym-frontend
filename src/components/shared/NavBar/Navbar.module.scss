.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: #1e1e2f;
  padding: 15px 20px;
  color: white;

  .logo {
    font-size: 1.5rem;
    font-weight: bold;
  }

  .actions {
    display: flex;
    gap: 20px;
  }

  .icon {
    font-size: 1.5rem;
    cursor: pointer;
    transition: color 0.3s ease;

    &:hover {
      color: #1cc88a;
    }
  }
}
