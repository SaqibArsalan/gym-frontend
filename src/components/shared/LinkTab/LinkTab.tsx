// Generated with util/create-component.js
import React from 'react';
import Tab from '@mui/material/Tab';

import { IconPositionEnum, ILinkTabProps } from './LinkTab.types';

import './LinkTab.scss';

const LinkTab: React.FC<ILinkTabProps> = (props: ILinkTabProps) => (
	<Tab {...props} />
);

LinkTab.defaultProps = {
	onClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
		event.preventDefault();
	},
	iconPosition: IconPositionEnum.start,
};

export default LinkTab;
