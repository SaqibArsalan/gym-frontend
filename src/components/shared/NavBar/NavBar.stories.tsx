// Generated with util/create-component.js
import React, { useState } from 'react';
import NavBar from './NavBar';
import LinkTab from '../LinkTab/LinkTab';
import { IconPositionEnum } from '../LinkTab';
// import SvgIconCustom from '../SvgIcon/SvgIcon';

export default {
	title: 'NavBar',
};

const iconImage = require('../../assets/images/cart-flatbed-boxes.svg');

// const icon1 = () => <SvgIconCustom src={iconImage} alt='icon-1' />;

const iconImage2 = require('../../assets/images/boxes-stacked.svg');

// const icon2 = () => <SvgIconCustom src={iconImage2} alt='icon-2' />;

export const WithMultipleTabs = () => (
	<NavBar>
		<LinkTab label='Link1' href='#' />
		<LinkTab
			label='Link2'
			href='#'
			iconPosition={IconPositionEnum.end}
		/>
		<LinkTab label='Link3' href='#' />
	</NavBar>
);

export const MultipleWithBorderRemoved = () => (
	<NavBar removeBorderBottom>
		<LinkTab
			label='Link2'
			href='#'
			iconPosition={IconPositionEnum.start}
		/>
		<LinkTab
			label='Custom Svg Icon'
			href='#'
			iconPosition={IconPositionEnum.start}
		/>
	</NavBar>
);

export const WithMultipleTabsAndCustomHandlers = () => {
	const [tabValue, setTabValue] = useState(4);
	return (
		<NavBar
			value={tabValue}
			onChange={(e, newValue) => {
				console.log('this is my custom handler', e);
				console.log('the new value here', newValue);
				setTabValue(newValue);
			}}
		>
			<LinkTab label='Link1' href='#' />
			<LinkTab
				label='Link2'
				href='#'
				iconPosition={IconPositionEnum.end}
			/>
			<LinkTab label='Link3' href='#' />
			<LinkTab label='Link4' href='#' />
			<LinkTab
				label='Link5'
				href='#'
				iconPosition={IconPositionEnum.end}
			/>
			<LinkTab label='Link6' href='#' />
		</NavBar>
	);
};
