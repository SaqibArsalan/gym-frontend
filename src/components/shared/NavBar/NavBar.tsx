// Generated with util/create-component.js
import React, { useState } from 'react';
import { Tabs } from '@mui/material';
import styled from '@emotion/styled';

import { INavBarProps } from './NavBar.types';

import './NavBar.scss';

const CustomizedTabs = styled(Tabs)(() => ({
	'.MuiTabs-indicator': {
		display: 'none',
	},
}));

const NavBar: React.FC<INavBarProps> = (props: INavBarProps) => {
	const [tabValue, setValue] = useState(props.value != null ? props.value : 0);

	const handleChange = (event: React.SyntheticEvent, newValue: number) => {
		setValue(newValue);
	};

	return props.removeBorderBottom ? (
		<CustomizedTabs
			{...props}
			value={props.value != null ? props.value : tabValue}
			onChange={props.onChange || handleChange}
			aria-label='nav tabs example'
		>
			{props.children}
		</CustomizedTabs>
	) : (
		<Tabs
			{...props}
			value={props.value != null ? props.value : tabValue}
			onChange={props.onChange || handleChange}
			aria-label='nav tabs example'
		>
			{props.children}
		</Tabs>
	);
};

NavBar.defaultProps = {
	removeBorderBottom: false,
	value: null,
	onChange: undefined,
};

export default NavBar;
