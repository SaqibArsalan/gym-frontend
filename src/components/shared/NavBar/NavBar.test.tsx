// Generated with util/create-component.js
import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import NavBar from './NavBar';
import { INavBarProps } from './NavBar.types';
import LinkTab from '../LinkTab/LinkTab';

describe('NavBar Component', () => {
	let wrapper: ReactWrapper<
		any,
		Readonly<{}>,
		React.Component<INavBarProps, {}, any>
	>;

	beforeEach(() => {
		wrapper = mount(
			<NavBar>
				<LinkTab label='link tab active' href='#' />
				<LinkTab label='link tab' href='#' />
			</NavBar>
		);
	});

	test('component should successfully render', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should render link tab inside navbar correctly', () => {
		const linkTab = wrapper.find('.MuiTabs-flexContainer .MuiTab-root').first();
		expect(linkTab).toHaveLength(1);
		expect(linkTab.text()).toBe('link tab active');
	});

	test('should test handleChange onClick and tabIndex of linkTab should update', () => {
		const linkTab = wrapper.find('.MuiTabs-flexContainer .MuiTab-root').last();

		expect(linkTab).toHaveLength(1);
		expect(linkTab.props().tabIndex).toBe(-1);

		const event: any = {};
		linkTab.simulate('click', event, 1);

		const updatedLinkTAb = wrapper
			.find('.MuiTabs-flexContainer .MuiTab-root')
			.last();
		expect(updatedLinkTAb.props().tabIndex).toBe(0);
	});
});
