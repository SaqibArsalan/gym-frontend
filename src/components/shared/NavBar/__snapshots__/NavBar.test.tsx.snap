// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component component should successfully render 1`] = `ReactWrapper {}`;
